keyboards/themadnoodle/noodlepad_micro/keymaps/default/keymap.c

```diff
@@ -3,6 +3,16 @@
 
 #include QMK_KEYBOARD_H
 
+enum custom_keycodes {
+#ifdef VIA_ENABLE    // If you are using VIA, you can use the same keycodes as the default keymap
+    L_IND = QK_KB_0, // Toggle the Layer Indicators Modes
+    L_CYC            // Cycle through the layers
+#else                // If you are not using VIA, you can use your own keycodes
+    L_IND = SAFE_RANGE,
+    L_CYC
+#endif
```

Comment:
`QK_KB_0` should be used at the kb level, not the keymap level. 
```suggestion
    L_IND = SAFE_RANGE, // Toggle the Layer Indicators Modes
    L_CYC               // Cycle through the layers
```

Reply:
Also, this was something that took a bunch of work to figure out with the help of the QMK Discord. This was the only way VIA would recognize the custom keycodes. 

It was a while ago, but I believe I tried to put it at the KB level, but it wasn't recognized by VIA either. Well, it would get recognized, but it wouldn't show up as the custom code. It would show up as another name, and i couldn't map to a switch. 

To help a noob out, should I remove everything below '/*Encoder Mapping*/' section and put it in a noodlepad_mico.c file along with the 'enum custom_keycodes' at the top?

If you can help me out, that'd be amazing because I'm really at a total loss.

Reply:
default keymap shouldn't have via enabled.

Reply:
@drashna AH I see! So you're saying just for the default keymap.